tests/pytests/test_resp3.py

```diff
@@ -708,3 +708,374 @@ def test_profile_child_itrerators_array():
+    ]
+  }
+
+  # Default FORMAT is STRING
+
+  # Test FT.SEARCH
+  res = env.cmd('FT.SEARCH', 'idx', '*', 'LIMIT', 0, 2)
+  res['results'].sort(key=lambda x: "" if x['extra_attributes'].get('num') == None else x['extra_attributes'].get('num'))
+  env.assertEqual(res, exp_string)
+
+  res = env.cmd('FT.SEARCH', 'idx', '*','LIMIT', 0, 2, 'FORMAT', 'STRING')
+  env.assertEqual(res, exp_string)
+
+  # Test FT.AGGREGATE
+  del exp_string['results'][0]['id']
+  del exp_string['results'][1]['id']
+  res = env.cmd('FT.AGGREGATE', 'idx', '*', 'LIMIT', 0, 2, 'SORTBY', 2, '@num', 'ASC', 'LOAD', '*')
+  env.assertEqual(res, exp_string)
+
+
+  #
+  # Return specific fields
+  #
+  exp_string = {
+    'attributes': [],
+    'error': [],
+    'total_results': ANY,
+    'format': 'STRING',
+    'results': [
+      {'id': 'doc1', 'extra_attributes': {'num': '1', 'other': 'fu'}, 'values': []},
+      {'id': 'doc2', 'extra_attributes': {'num': '2', 'other': 'bur'}, 'values': []},
+    ]
+  }
+
+  # Test FT.SEARCH
+  res = env.cmd('FT.SEARCH', 'idx', '*', 'LIMIT', 0, 2, 'RETURN', 2, 'num', 'other')
+  env.assertEqual(res, exp_string)
+
+  # Test FT.AGGREGATE
+  del exp_string['results'][0]['id']
+  del exp_string['results'][1]['id']
+  res = env.cmd('FT.AGGREGATE', 'idx', '*', 'LIMIT', 0, 2, 'LOAD', 2, 'num', 'other', 'SORTBY', 2, '@num', 'ASC')
+  env.assertEqual(res, exp_string)
+
+
+def testExpandJsonVector():
+  ''' Test returning values for VECTOR in expanded format (raw RESP3 instead of stringified JSON) '''
+  env = Env(protocol=3, moduleArgs='DEFAULT_DIALECT 2')
+  conn = getConnectionByEnv(env)
+  conn.execute_command('FT.CREATE', 'idx', 'ON', 'JSON',
+                      'SCHEMA', '$.v', 'AS', 'vec', 'VECTOR', 'FLAT', '6', 'TYPE', 'FLOAT32', 'DIM', '3','DISTANCE_METRIC', 'L2',
+                      '$.num', 'AS', 'num', 'NUMERIC')
+
+
+  with env.getClusterConnectionIfNeeded() as r:
+    r.execute_command('json.set', 'doc1', '$', '{"v":[1,2,3],"num":1}')
+    r.execute_command('json.set', 'doc2', '$', '{"v":[4,2,0],"num":2}')
+
+  exp_string = {
+    'attributes': [],
+    'error': [],
+    'total_results': 2,
+    'format': 'STRING',
+    'results': [
+      {'id': 'doc1', 'extra_attributes': {'__vec_score': '6.70958423615', '$': '{"v":[1,2,3],"num":1}'}, 'values': []},
+      {'id': 'doc2', 'extra_attributes': {'__vec_score': '12.7095842361', '$': '{"v":[4,2,0],"num":2}'}, 'values': []}
+    ]
+  }
+  
+  exp_expand = {
+    'attributes': [],
+    'error': [],
+    'total_results': 2,
+    'format': 'EXPAND',
+    'results': [
+      {'id': 'doc1', 'extra_attributes': {'__vec_score': 6.7095842361450195, '$': [{'v': [1, 2, 3], 'num': 1}]}, 'values': []},
+      {'id': 'doc2', 'extra_attributes': {'__vec_score': 12.70958423614502, '$': [{'v':[4, 2, 0], 'num': 2}]}, 'values': []},
+    ]
+  }
+  
+  cmd = ['FT.SEARCH', 'idx', '*=>[KNN 2 @vec $B]', 'PARAMS', '2', 'B', '????????????']
```

Comment:
use numpy for real vector. you can send the zero vector so you'll have real L2 distance

Reply:
Is it required here? There are such simple queries in test_vecsim,
e.g., https://github.com/RediSearch/RediSearch/blob/2d1c0d537116eac0aff1f35a4475816c8f78a70d/tests/pytests/test_vecsim.py#L301